build/index.js

```diff
@@ -266,12 +267,40 @@ function logSuccess(command, timeLapsed) {
     logMain(message);
 }
 
+function checkVersions(metaManifest) {
+    const thisNPM = process.platform === "win32" ? "npm.cmd" : "npm";
+    let exitCode = 0;
+    const nodeMajorVersion = parseInt(process.version.slice(1, 3), 10);
+    if (nodeMajorVersion < metaManifest.requiredNodeMajorVersion) {
+        console.error(
+            `⚠️ Node.js v${metaManifest.requiredNodeMajorVersion} or later ` +
+                `required, you are on ${process.version}`
+        );
+        exitCode = 1;
+    }
+    const npmVersion = execSync(`${thisNPM} --version`, { encoding: 'utf-8' });
+    const npmMajorVersion = npmVersion.match(/^(\d+)\.\d+/)[0];
+    if (npmMajorVersion < metaManifest.requiredNPMMajorVersion) {
+            console.error(
+                `⚠️ NPM ${metaManifest.requiredNPMMajorVersion} or later ` +
+                `required, you are on ${npmMajorVersion}`
+            );
+            exitCode = 1;
+    }
+    return exitCode;
+}
+
 async function main() {
     const { scriptPath, command, start } = parseCommandLine();
     if (!command) {
         showUsage(scriptPath);
         return 1;
     }
+    const metaManifest = parseManifest("build/packages.json");
+    const exitCode = checkVersions(metaManifest);
+    if (exitCode == 1 ) {
```

Comment:
I did suggest to instead catch an error from `checkVersions` here, but if you'd be very opposed to this, I think this double hard coding of `1` seems wrong, I'd suggest: `if (exitCode !== 0) return exitCode;`